src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -36,6 +38,21 @@ public ODataSerializerContext()
         /// </param>
         /// <remarks>This constructor is used to construct the serializer context for writing nested and expanded properties.</remarks>
         public ODataSerializerContext(ResourceContext resource, SelectExpandClause selectExpandClause, IEdmProperty edmProperty)
+            : this(resource, selectExpandClause, edmProperty, null, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ODataSerializerContext"/> class.
+        /// </summary>
+        /// <param name="resource">The resource whose navigation property is being expanded.</param>
+        /// <param name="selectExpandClause">The <see cref="SelectExpandClause"/> for the navigation property being expanded.</param>
+        /// <param name="edmProperty">The complex property being nested or the navigation property being expanded.
+        /// If the resource property is the dynamic complex, the resource property is null.
+        /// </param>
+        /// <param name="queryOptions">The <see cref="ODataQueryOptions"/> for the navigation property being expanded.</param>
+        /// <param name="expandedItem">The <see cref="ExpandedNavigationSelectItem"/> for the navigation property being expanded.></param>
+        public ODataSerializerContext(ResourceContext resource, SelectExpandClause selectExpandClause, IEdmProperty edmProperty, ODataQueryOptions queryOptions, ExpandedNavigationSelectItem expandedItem)
```

Comment:
>queryOptions [](start = 147, length = 12)

what's the relationship between the "selectExpandClause" and the QueryOption?

Reply:
queryoptions has SelectExpandQueryOption which contains the SelectExpandClause. The plan is to remove SelectExpandClause when we do a major release.